e2etests/node_scan_rhcos_test.go

```diff
@@ -0,0 +1,166 @@
+					RhelComponents: []*v1.RHELComponent{
+						{
+							Id:        int64(1),
+							Name:      "libksba",
+							Namespace: "rhel:8",
+							Version:   "1.3.5-7.el8",
+							Arch:      "x86_64",
+							Module:    "", // must be empty, otherwise scanner does not return any vulns
+							// From: https://www.redhat.com/security/data/metrics/repository-to-cpe.json
+							// "rhel-8-for-x86_64-appstream-rpms": {"cpes": ["cpe:/a:redhat:enterprise_linux:8::appstream", "cpe:/a:redhat:rhel:8.3::appstream"]},
+							// "rhel-8-for-x86_64-baseos-rpms": {"cpes": ["cpe:/o:redhat:enterprise_linux:8::baseos", "cpe:/o:redhat:rhel:8.3::baseos"]}
+							Cpes: []string{
+								"cpe:/a:redhat:enterprise_linux:8::appstream", "cpe:/a:redhat:rhel:8.3::appstream",
+								"cpe:/a:redhat:enterprise_linux:8::baseos", "cpe:/a:redhat:rhel:8.3::baseos",
+							},
+							AddedBy: "",
+						},
+						{
+							Id:        int64(2),
+							Name:      "tar",
+							Namespace: "rhel:8",
+							Version:   "1.27.1.el8",
+							Arch:      "x86_64",
+							Module:    "",
+							Cpes: []string{
+								"cpe:/a:redhat:enterprise_linux:8::appstream", "cpe:/a:redhat:rhel:8.3::appstream",
+								"cpe:/a:redhat:enterprise_linux:8::baseos", "cpe:/a:redhat:rhel:8.3::baseos",
+							},
+							AddedBy: "",
+						},
+					},
+					LanguageComponents: nil,
+				},
+			},
+			responseContains: &v1.GetNodeVulnerabilitiesResponse{
+				InventoryFeatures: []*v1.Feature{
+					{
+						Name:            "libksba",
+						Version:         "1.3.5-7.el8.x86_64",
+						Vulnerabilities: []*v1.Vulnerability{vulnLibksba},
+					},
+					{
+						Name:            "tar",
+						Version:         "1.27.1.el8.x86_64",
+						Vulnerabilities: []*v1.Vulnerability{vulnTar},
+					},
+				},
+			},
+		},
+	}
+
+	for i, c := range cases {
+		t.Run(fmt.Sprintf("case-%d", i), func(t *testing.T) {
+			c := c
+			resp, err := client.GetNodeVulnerabilities(context.Background(), c.request)
+			require.NoError(t, err)
+			for _, expectedFeat := range c.responseContains.InventoryFeatures {
+				var found bool
+				for _, gotFeat := range resp.InventoryFeatures {
+					if expectedFeat.GetName() == gotFeat.GetName() && expectedFeat.GetVersion() == gotFeat.GetVersion() {
+						found = true
+						assert.NotNil(t, gotFeat)
+						assert.NotNil(t, gotFeat.Vulnerabilities, "Expected to find vulnerabilities for %s:%s", expectedFeat.GetName(), expectedFeat.GetVersion())
+						t.Logf("gotFeat.InventoryFeatures: %+v\n", resp.InventoryFeatures)
+						assertContainsVuln(t, gotFeat.Vulnerabilities, expectedFeat.Vulnerabilities)
+					}
+				}
+				assert.Truef(t, found, "expected to find feat '%s:%s' in the reply, but got none. Features in the reply: %+v", expectedFeat.GetName(), expectedFeat.GetVersion(), resp.InventoryFeatures)
+			}
+			assert.Equal(t, c.responseContains.Notes, resp.Notes)
+		})
+	}
+}
+
+func assertContainsVuln(t *testing.T, foundVulns, expectedContains []*v1.Vulnerability) {
```

Comment:
1. Please use getters inside.
2. Please assert that lengths of `foundVulns` and `expectedContains` are equal. Unless you mean that `expectedContains` can be a subset of `foundVulns`.

Reply:
Yes, I meant to assert that the expected vulnerability exists in the `foundVulns`.  Maybe I rename this function to make it clear.